eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPrePublisher.scala

```diff
@@ -64,22 +64,33 @@ object ReplaceableTxPrePublisher {
     def txInfo: ReplaceableTransactionWithInputInfo
     def updateTx(tx: Transaction): ReplaceableTxWithWitnessData
   }
-  /** Replaceable transaction for which we may need to add wallet inputs. */
+  /** Replaceable transaction for which we may need to add wallet inputs and outputs. */
```

Comment:
All the changes made to this file seem unnecessary: for all of these transactions, we have exactly one input and one output that don't belong to the wallet (and must use the same index for HTLC txs because our peer used `SIGHASH_SINGLE | SIGASH_ANYONECANPAY`). We shouldn't need to explicitly track anything, we can do this on-the-fly when evaluating the final transaction.

Reply:
Explicit tracking of our wallet inputs and outputs is a small change that makes reasoning about the code simpler (since the PSBT signing code needs to verify such inputs and outputs) and has very little impact on the rest f the code base, I' d rather keep it unless you really think that it should be removed (I agree that this "eclair manages bitcoin keys" mode will not be used very often and we should minimize its impact on our code base, maybe I've missed something...).

Reply:
> Explicit tracking of our wallet inputs and outputs is a small change that makes reasoning about the code simpler

I disagree...I think it's unnecessary complexity: it adds quite a lot of code that isn't easy at all to reason about (since those values are set and read across different components that make regular changes to the structure of the transactions and need to make sure they also correctly update those new fields). The various transaction publishers are already quite complex (mostly because we need to work around some bitcoind limitations, which I hope we can simplify in the future) and are extremely sensitive, so I'd rather minimize changes and optimize for simplicity. Always using the first position in the transaction for our non-wallet input and output is the simplest thing I can think of, don't you agree?

Reply:
Changed in https://github.com/ACINQ/eclair/pull/2613/commits/aecae9aded205cb0bbacca99a3a23b520821fb4e